rust/connlib/tunnel/src/control_protocol/gateway.rs

```diff
@@ -96,7 +96,8 @@ where
                     tracing::trace!("new_data_channel_open");
                     Box::pin(async move {
                         {
-                            let Some(iface_config) = tunnel.iface_config.read().clone() else {
+                            let Some(iface_config) = tunnel.iface_config.read().await.clone()
+                            else {
```

Comment:
What is this weird formatting? Does `rustfmt` do this?

Reply:
*sigh* rustfmt doesn't get along with `let..else` yet and I'm used to depend on that for my indentation.